internal/backend/remote-state/gcs/backend_test.go

```diff
@@ -43,40 +43,115 @@ const (
-	t.Parallel()
+func preCheckTestAcc(t *testing.T) {
+	if v := os.Getenv("TF_ACC"); v == "" {
+		t.Skip("TF_ACC must be set to run acceptance tests, as they provision real resources")
+	}
+}
 
-	cases := []struct {
-		prefix        string
-		name          string
-		wantStateFile string
-		wantLockFile  string
-	}{
-		{"state", "default", "state/default.tfstate", "state/default.tflock"},
-		{"state", "test", "state/test.tfstate", "state/test.tflock"},
-		{"state", "test", "state/test.tfstate", "state/test.tflock"},
-		{"state", "test", "state/test.tfstate", "state/test.tflock"},
-	}
-	for _, c := range cases {
-		b := &Backend{
-			prefix: c.prefix,
-		}
+func preCheckEnvironmentVariables(t *testing.T) {
 
-		if got := b.stateFile(c.name); got != c.wantStateFile {
-			t.Errorf("stateFile(%q) = %q, want %q", c.name, got, c.wantStateFile)
-		}
+	// TODO - implement a separate function specific to credentials which will iterate through a list of ENV names and return first value found.
 
-		if got := b.lockFile(c.name); got != c.wantLockFile {
-			t.Errorf("lockFile(%q) = %q, want %q", c.name, got, c.wantLockFile)
+	credentials := []string{
+		"GOOGLE_BACKEND_CREDENTIALS",
+		"GOOGLE_CREDENTIALS",
+	}
+	credsFound := false
+	for _, name := range credentials {
+		v := os.Getenv(name)
+		if v != "" {
+			credsFound = true
+			break
 		}
 	}
+	if !credsFound {
+		// Skipping tests because hashicorp/terraform repo doesn't have credentials set up.
+		// In future we should enable tests to run automatically, and make this code fail when no creds are set.
+		t.Skip("credentials need to be provided via GOOGLE_BACKEND_CREDENTIALS or GOOGLE_CREDENTIALS environment variable but neither is set")
+	}
+}
+
+func TestAccBackendConfig_credentials(t *testing.T) {
+	// Cannot use t.Parallel() due to t.Setenv
+	preCheckTestAcc(t)
+	preCheckEnvironmentVariables(t)
+
+	credentials := os.Getenv("GOOGLE_BACKEND_CREDENTIALS")
+	if credentials == "" {
+		credentials = os.Getenv("GOOGLE_CREDENTIALS")
+	}
+	if credentials == "" {
+		t.Fatalf("unable to access credentials from the test environment")
+	}
+
+	t.Setenv("GOOGLE_BACKEND_CREDENTIALS", "") // unset value
+	t.Setenv("GOOGLE_CREDENTIALS", "")         // unset value
+
+	cases := map[string]struct {
+		config map[string]interface{}
+		envs   map[string]string
+		want   string
+	}{
+		"empty credentials in config doesn't affect use of GOOGLE_BACKEND_CREDENTIALS": {
+			config: map[string]interface{}{
+				"bucket":      "tf-test-testaccbackendconfig_credentials_1",
+				"credentials": "",
+			},
+			envs: map[string]string{
+				"GOOGLE_BACKEND_CREDENTIALS": credentials,
+			},
+		},
+		"empty credentials in config doesn't affect use of GOOGLE_CREDENTIALS": {
+			config: map[string]interface{}{
+				"bucket":      "tf-test-testaccbackendconfig_credentials_2",
+				"credentials": "",
+			},
+			envs: map[string]string{
+				"GOOGLE_CREDENTIALS": credentials,
+			},
+		},
+		// Uncomment below for sanity checking
```

Comment:
It's probably best not to merge in lines of commented out code – if this is a purposeful test it would be better to formulate it in a way that a failing test isn't the desired outcome.

Reply:
That true - I'll change it to a TODO comment for the future to add a test case for X and to make changes to the shared backend package to enable that